browser/playlist/test/playlist_browsertest.cc

```diff
@@ -358,6 +359,31 @@ IN_PROC_BROWSER_TEST_F(PlaylistBrowserTest, PlaylistTabHelper) {
       [&]() { return playlist_tab_helper->found_items().size() == 0; }));
 }
 
+IN_PROC_BROWSER_TEST_F(PlaylistBrowserTest,
```

Comment:
How about using `MockPlaylistServiceObserver`, and defining expectations (`EXPECT_CALL`) on `OnMediaFilesUpdated` instead?

Reply:
I just tested it, but it doesn't seem that OnMediaFilesUpdate() is filed when a node is removed. I think this is because we're observing mutation of src attribute